### PR TITLE
Auth UI v2 - Add enter recovery code page

### PR DESCRIPTION
**Pull request by IniZio**
ref #3580

dependencies:
- [x] #3682

Screenshot 2024-01-25 at 09-53-33 Authgear

### CODE REVIEW

**Comment by tung2744** [comment]
It seems still need a  rebase :sadfrog:

**Comment by IniZio** [comment]
Changed base branch and rebased

**Comment by tung2744** [comment]
I found that in this screen, the otp alterernative "Send code via whatsapp" is a bit weird, because it looks like it will send the recovery code to you by whatsapp 🤔

File: resources/authgear/templates/en/translation.json [hunk 1]
```diff
@@ -966,6 +966,7 @@
   "v2-error-server": "Server encountered an internal error.",
   "v2-error-screen-title": "Unexpected Issues",
   "v2-error.error-password-policy-violated": "Invalid password format.",
+  "v2-error-phone-number-format": "ncorrect phone number format for the selected country.",
```
Comment (tung2744) [inline resources/authgear/templates/en/translation.json:969]: typo

**Comment by IniZio** [comment]
Previously there will always be a "... via XXX instead" copywriting. New UI does look cleaner, guess maybe can choose a different term for OTP code from recovery code 🤔

File: resources/authgear/templates/en/web/authflowv2/setup_oob_otp.html [hunk 2]
```diff
@@ -39,6 +38,18 @@ <h2 class="screen-description">
...
+        {{ if not $display_otp_input_error }}
+          {{ template "authflowv2/__alert_message.html"
+            (dict
+              "Type" "error"
+              "Classname" "max-tablet:hidden mt-4"
+              "Message" (include "authflowv2/__error.html" .)
+            )
+          }}
+        {{ end }}
+      {{ end }}
```
Comment (tung2744) [inline resources/authgear/templates/en/web/authflowv2/setup_oob_otp.html:52]: Where did you find this design?

btw, it looks like `__alert_message` should be something global in all screen, which displays any error not handled in that screen. Maybe we should create an issue to review the usage of `__alert_message` in other screens too?

File: resources/authgear/templates/en/translation.json (hunk 1 above)
Comment (IniZio) [reply resources/authgear/templates/en/translation.json]: Fixed in 07d9746dcca35529d34a99ae42726ac3ab8acb21

File: resources/authgear/templates/en/web/authflowv2/setup_oob_otp.html (hunk 2 above)
Comment (IniZio) [reply resources/authgear/templates/en/web/authflowv2/setup_oob_otp.html]: Desktop: https://www.figma.com/file/LRE8EfCaTfYxC0lk5AeMmE/New-Auth-UI?type=design&node-id=1148-41148&mode=design&t=PDG0RvzBvlSXJgDi-4
Mobile: https://www.figma.com/file/LRE8EfCaTfYxC0lk5AeMmE/New-Auth-UI?type=design&node-id=822-16106&mode=design&t=PDG0RvzBvlSXJgDi-4

You still need to tell `__alert_message` what error is ignored, which I think is screen-specific?

Comment (tung2744) [reply resources/authgear/templates/en/web/authflowv2/setup_oob_otp.html]: Opened https://github.com/authgear/authgear-server/issues/3698

**Pull request merged**